Assets/Scripts/Back-end/State machine/StateController.cs

```diff
@@ -0,0 +1,47 @@
+﻿using UnityEngine.AI;
+using UnityEngine;
+
+/// <summary>
+/// The controller of the AI, this is the brain that controlls the AI.
+/// </summary>
+[RequireComponent(typeof(NavMeshAgent))]
+public class StateController : MonoBehaviour
+{
+    // The NavMesh controller component.
+    private NavMeshAgent _navMeshAgent;
+
+    // The current state in which the AI currently is.
+    [SerializeField] private State _currentState = null;
+
```

Comment:
Put [SerializeField] above the parameter (not in conventions, but Casper recommends)